jaxrs/src/test/java/org/killbill/billing/jaxrs/resources/TestOverdueResource.java

```diff
@@ -0,0 +1,146 @@
+import org.killbill.billing.jaxrs.util.Context;
+import org.killbill.billing.jaxrs.util.JaxrsUriBuilder;
+import org.killbill.billing.overdue.api.OverdueApi;
+import org.killbill.billing.overdue.api.OverdueConfig;
+import org.killbill.billing.overdue.config.DefaultOverdueConfig;
+import org.killbill.billing.payment.api.InvoicePaymentApi;
+import org.killbill.billing.payment.api.PaymentApi;
+import org.killbill.billing.util.api.AuditUserApi;
+import org.killbill.billing.util.api.CustomFieldUserApi;
+import org.killbill.billing.util.api.TagUserApi;
+import org.killbill.billing.util.callcontext.TenantContext;
+import org.killbill.clock.Clock;
+import org.killbill.xmlloader.XMLLoader;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.google.common.io.Resources;
+
+import static org.mockito.Mockito.*;
+
+public class TestOverdueResource extends TestJaxRsResourceBase {
+
+    private final ServletRequest servletRequest = mock(ServletRequest.class);
+    private final TenantContext tenantContext = mock(TenantContext.class);
+
+    private final JaxrsUriBuilder jaxrsUriBuilder = mock(JaxrsUriBuilder.class);
+    private final TagUserApi tagUserApi = mock(TagUserApi.class);
+    private final CustomFieldUserApi customFieldUserApi = mock(CustomFieldUserApi.class);
+    private final AuditUserApi auditUserApi = mock(AuditUserApi.class);
+    private final AccountUserApi accountUserApi = mock(AccountUserApi.class);
+    private final PaymentApi paymentApi = mock(PaymentApi.class);
+    private final InvoicePaymentApi invoicePaymentApi = mock(InvoicePaymentApi.class);
+    private final OverdueApi overdueApi = mock(OverdueApi.class);
+    private final Clock clock = mock(Clock.class);
+    private final Context context = mock(Context.class);
+
+
+    private OverdueResource newOverdueResource() {
+        return new OverdueResource(
+                jaxrsUriBuilder,
+                tagUserApi, customFieldUserApi, auditUserApi, accountUserApi,
+                paymentApi, invoicePaymentApi, overdueApi,
+                clock, context);
+    }
+
+
+    protected DefaultOverdueConfig getOverdueConfig(final String name) throws Exception {
+        return XMLLoader.getObjectFromString(Resources.getResource("org/killbill/billing/jaxrs/resources/overdue/" + name).toExternalForm(), DefaultOverdueConfig.class);
+    }
+
+    /**
+     * Represent invalid overdue that cannot marshalled into JSON. This XML contains the same value as defined in:
+     * https://github.com/killbill/killbill/issues/1497
+     *
+     * This XML causing NullPointerException when JSON marshalling because it missing
+     * 'timeSinceEarliestUnpaidInvoiceEqualsOrExceeds' element.
+     */
+    @Test(groups = "fast", description = "https://github.com/killbill/killbill/issues/1497")
+    public void testGetOverdueConfigJson_noTimeSinceEarliest() throws Exception {
+        final OverdueConfig overdueConfig = getOverdueConfig("overdue_no-TimeSinceEarliest.xml");
```

Comment:
Your naming conventions are a bit hard to follow, this is a mix of camel/snake cases and `-`... Can we settle on something a bit simpler (e.g under `beatrix/src/test/resources/org/killbill/billing/beatrix/`)